### ISSUE TITLE
Insane cpu usage and ram!

### ISSUE THREAD

**Issue by saif-ellafi** [jstaf/onedriver#203]
Hi! was debugging some performance issues on my pc. Not sure if related but I found out this process on top of the list! gives me some concerns.

This is taking almost 4GB of Ram and about +50% my total's processors load

![image](https://user-images.githubusercontent.com/27952699/141698610-5c5f04aa-6d07-4cdd-9bc2-d1a1ec75697e.png)

**Comment by abraunegg** [comment]
@saif-ellafi 
Whilst I will let @jstaf respond in better detail, what I do notice here is `baloo_file_extractor` is running.

In the [past](https://github.com/abraunegg/onedrive/issues/824#issuecomment-603469431) it has been seen on a number of occasions the 'Dolphin File Indexer' (which is the baloo_file_extractor process) constantly accesses and updates the files 'modified' timestamp. An indexing application should never update the modified timestamp of a file to track when it was last indexed ... this is very poor way to do this.

Essentially, every time this process runs, it can trigger an update - which, depending on how 'onedriver' works, will account for what is going on.

Your options here would be to stop using the 'Dolphin File Indexer' and see if your system calms down from all.

**Comment by saif-ellafi** [comment]
Thank you, for now I switched to web based onedrive and moved to an alternative cloud provider that has Linux support.
I really appreciate your work here it is tremendous, and a shame by Microsoft for not doing it, so I feel bad about pointing here linux specific difficulties :) So thanks again for doing this.

Will investigate what is this Dolphin File Indexer and what are the the side effects of disabling it.

**Comment by jstaf** [comment]
Yeah this is probably KDE's default baloo file indexer. You can either disable it completely via `balooctl suspend` + `balooctl disable` or have it exclude your mountpoint folder via https://community.kde.org/Baloo/Configuration#Exclude_Folders

There's also a fix for some of the CPU usage in the next version - I've had at least one report that it's made a huge improvement: https://github.com/jstaf/onedriver/issues/196#issuecomment-963554401

**Comment by saif-ellafi** [comment]
Thank you. Any brief explanation on why the baloo indexer is not an issue for other synchronizers? (Dropbox, mega, etc)

**Comment by jstaf** [comment]
Because those synchronizers are "syncing files". Everything has been synced ahead of time.

There is no "syncing" happening here- when you try to open a file, onedriver tries to do an on-demand download of the file and present it to the program that opened it. So by "indexing" your mountpoint, you are forcing the filesystem to download every file you have while the indexer is running.

**Comment by saif-ellafi** [comment]
I see. I wonder if there is anything on a filesystem level where a flag can be set to disable indexing on there? Or maybe somehow can onedriver add itself to exceptions?

**Comment by saif-ellafi** [comment]
Also on my screenshot, the baloo_file_extractor was not always on top. Only sporadically. onedriver on the other hand was always on top of everything else.

**Comment by NICHOLAS85** [comment]
Keep in mind having file previews enabled as well in OneDrive folder will cause increased cpu usage as the files are downloaded to show the preview

**Comment by jstaf** [comment]
> I see. I wonder if there is anything on a filesystem level where a flag can be set to disable indexing on there? Or maybe somehow can onedriver-launcher add itself to exceptions?

Yeah, see my comment about excluding specific folders from Baloo. 

Likewise, there is also a massive improvement to CPU usage in the latest version on the master branch.

**Comment by saif-ellafi** [comment]
Thank you. Will try again once the master branch gets released!

**Comment by jstaf** [comment]
Version 0.12.0 is now released for all of the build systems I manage (might need to wait on the AUR maintainer if you use arch btw). Try again and let me know if you see an improvement.

**Comment by saif-ellafi** [comment]
Thanks @jstaf will check it out this evening. I realized I do need baloo indexer, because I have plenty of files and use extensively the search in there.
However I might be able to disable content indexing as I mostly search by filename.

**Comment by saif-ellafi** [comment]
None of the builds in AUR are still available, waiting a bit more :)

**Comment by saif-ellafi** [comment]
Hello! I have just installed this, and however I have moved away from OneDrive a bit, I got to test this. Unfortunately from my latest time, I have much less files in OneDrive now and smaller, and also deactivated baloo "content indexer" (I only index file names), since anyways it worked awfully bad (bringing garbage results in system searches).

Anyways, with the current build and onedriver running on the background, it looks like this (miles better!)
![image](https://user-images.githubusercontent.com/27952699/145657233-261c32f6-5aa4-4eec-b8c2-cddb80dfc3d1.png)
